`request_kwargs_matcher` not working

### Describe the bug

I'm trying to mock a request that passes request arguments. In this case, `allow_redirects=False`. I'm reading the document but I can't seem to get it working: https://github.com/getsentry/responses#request-keyword-arguments-matcher

Am I doing something wrong or is this a bug?

### Additional context

_No response_

### Version of `responses`

0.22.0

### Steps to Reproduce

```python
import requests
import responses
from responses import matchers

@responses.activate(assert_all_requests_are_fired=True)
def test_responses(self):
    url = "https://example.com/"

    responses.get(
        url,
        match=[matchers.request_kwargs_matcher({"allow_redirects": False})],
    )

    requests.get(url, allow_redirects=False)
```

### Expected Result

Test should pass.

### Actual Result

Test fails with the following error:

```
E           requests.exceptions.ConnectionError: Connection refused by Responses - the call doesn't match any registered mock.
E
E           Request:
E           - GET https://example.com/
E
E           Available matches:
E           - GET https://example.com/ Arguments don't match: {} doesn't match {allow_redirects: False}
```

--- a/README.rst
+++ b/README.rst
@@ -504,6 +504,7 @@ Request Keyword Arguments Matcher
 
 To validate request arguments use the ``matchers.request_kwargs_matcher`` function to match
 against the request kwargs.
+Only following arguments are supported: ``timeout``, ``verify``, ``proxies``, ``stream``, ``cert``.
 
 Note, only arguments provided to ``matchers.request_kwargs_matcher`` will be validated.
 

